TundraTD/Assets/Scripts/City/Building/TowerPurchaseMenu.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using Spells;
+using UnityEngine;
+
+namespace City.Building
+{
+    public class TowerPurchaseMenu : MonoBehaviour
+    {
+        public int SelectedTowerID { get; set; }
+        
+        private void Start()
+        {
+            gameObject.SetActive(false);
+        }
+        
+        public void BuildTower(string elementName)
+        {
+            BasicElement element = TransformStringIntoBasicElement(elementName);
+            if (element == BasicElement.None)
+                throw new Exception("No element found for tower"); 
+            Architect.BuildNewTower(SelectedTowerID, element);
+            
+            BasicElement TransformStringIntoBasicElement(string rawElementName)
```

Comment:
I suppose it's a bad decision to load elements such as that way. Firstly, you're making extensibility of the application lower. Secondly, string comparison using lower/upper case allocates more memory and spends more processor time which may affect another optimization issues. And thirdly you could use `Enum.Parse<MagicElement>(value)` which could solve this problem for you (I hope it's available in that ancient version of sharp that is used in Unity)
```suggestion
            BasicElement element = Enum.Parse<BasicElement>(elementName, true);
```